core/scripts/local/BatchGasEstimation.js

```diff
@@ -0,0 +1,208 @@
+  console.log("batchCommit used", result.receipt.gasUsed, "gas to fit", requestNum, "in one tx.");
+  results["batchCommit"] = { number: requestNum, gasUsed: result.receipt.gasUsed };
+};
+
+// Test how many reveals can be placed within one block. The process for requesting and committing
+// are done linearly (not batched) so that the test for revels is not impacted by this.
+const cycleReveal = async (voting, identifier, time, requestNum, registeredDerivative, voter) => {
+  await generatePriceRequests(requestNum, voting, identifier, time, registeredDerivative);
+
+  await moveToNextRound(voting);
+
+  const salts = {};
+  const price = getRandomSignedInt();
+
+  // Generate Commitments. We will use single commit so no upper bound from the previous test
+  for (var i = 0; i < requestNum; i++) {
+    const salt = getRandomUnsignedInt();
+    const hash = web3.utils.soliditySha3(price, salt);
+    await voting.commitVote(identifier, time + i, hash, { from: voter });
+    salts[i] = salt;
+  }
+
+  // Advance to reveal phase
+  await moveToNextPhase(voting);
+
+  let reveals = [];
+  for (var i = 0; i < requestNum; i++) {
+    reveals[i] = { identifier: identifier, time: time + i, price: price.toString(), salt: salts[i].toString() };
+  }
+  const result = await voting.batchReveal(reveals, { from: voter });
+  console.log("batchReveal used", result.receipt.gasUsed, "gas to fit", requestNum, "in one tx.");
+  results["batchReveal"] = { number: requestNum, gasUsed: result.receipt.gasUsed };
+};
+
+// Test how many claims can be fit within one block. Generation of price requests, commits and
+// reveals are all done linearly.
+const cycleClaim = async (voting, identifier, time, requestNum, registeredDerivative, voter) => {
+  await generatePriceRequests(requestNum, voting, identifier, time, registeredDerivative);
+
+  await moveToNextRound(voting);
+
+  const salts = {};
+  const price = getRandomSignedInt();
+
+  for (var i = 0; i < requestNum; i++) {
+    const salt = getRandomUnsignedInt();
+    const hash = web3.utils.soliditySha3(price, salt);
+    await voting.commitVote(identifier, time + i, hash, { from: voter });
+    salts[i] = salt;
+  }
+
+  // Advance to reveal phase
+  await moveToNextPhase(voting);
+
+  for (var i = 0; i < requestNum; i++) {
+    await voting.revealVote(identifier, time + i, price, salts[i], { from: voter });
+  }
+
+  // Finally generate the batch rewards to retrieve
+  let roundId = await voting.getCurrentRoundId();
+
+  await moveToNextRound(voting);
+
+  let pendingRequests = [];
+  for (var i = 0; i < requestNum; i++) {
+    pendingRequests[i] = { identifier: identifier, time: time + i };
+  }
+  const result = await voting.retrieveRewards(voter, roundId, pendingRequests, { from: voter });
+  console.log("retrieveRewards used", result.receipt.gasUsed, "gas to fit", requestNum, "in one tx.");
+  results["retrieveRewards"] = { number: requestNum, gasUsed: result.receipt.gasUsed };
```

Comment:
Just curious: is there another value, like `result.receipt.gasUsed` that indicates how much gas was *required* to make the call, rather than what was used?

Reply:
There is a way to do this using `web3.eth.getGasPrice` and `estimateGas` however my research shows that it is not necessarily as accurate. Also at this point in the script it would not speed it up or result in any meaningful change as the main slowdown comes from the preceding steps in getting ready for the batch functionality like requesting prices, committing ect.